ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -113,16 +113,17 @@ class MySqlNamedBlobDb implements NamedBlobDb {
    */
   // @formatter:off
   private static final String LIST_QUERY_V2 = String.format(""
```

Comment:
I take a look at our original code, I suppose the reason we have such complex query is because we want to make sure the blobId we queried is match with the latest version.
For example 
blobName1 blobId2 ver1
blobName2 blobId1 ver2
with your query, it will return the max verison which is ver2, but we can't garentee it will return the blobId1. 
But I checked the list query, at handler level we only need the blobName, so I suggest instead of return the full namedBlobRecord, we should return the blobName, in that case, we don't need to worry about which blobId we returned.

Reply:
I don't understand this example. If i understand correctly, we need the latest version of a blob that is Ready. If it is deleted, then return nothing. Else return it. Am i missing something /